fe/fe-core/src/main/java/org/apache/iceberg/hive/IcebergHiveCatalog.java

```diff
@@ -354,20 +356,30 @@ public boolean dropTable(TableIdentifier identifier, boolean purge) {
 
     @Override
     public void deleteUncommittedDataFiles(List<String> fileLocations) {
-        fileLocations.forEach(this::deletePath);
+        for (String location : fileLocations) {
+            fileIO.deleteFile(location);
+        }
     }
 
-    // TODO(stephen): use CachingFileIO to access file system After adaptation
-    private void deletePath(String location) {
-        Path path = new Path(location);
-        URI uri = path.toUri();
-        try {
-            FileSystem fileSystem = FileSystem.get(uri, conf);
-            fileSystem.delete(path, true);
-        } catch (IOException e) {
-            LOG.error("Failed to delete location {}", location, e);
-            throw new StarRocksConnectorException("Failed to delete location %s. msg: %s", location, e.getMessage());
+    private void deleteTableFolder(FileIO io, TableMetadata metadata) {
+        String location = metadata.location();
+
+        // delete metadata folder
+        String metadataLocation = metadata.properties().get(TableProperties.WRITE_METADATA_LOCATION);
+        if (metadataLocation == null) {
+            metadataLocation = DEFAULT_METADATA_FOLDER_NAME;
         }
+        io.deleteFile(location + "/" + metadataLocation);
```

Comment:
maybe you need to check this metadataLocation.

Reply:
done